crates/context/config/src/client/env/config/mutate.rs

```diff
@@ -56,28 +62,75 @@ impl<'a> Method<Near> for Mutate<'a> {
-        // sign the params, encode it and return
-        // since you will have a `Vec<Felt>` here, you can
-        // `Vec::with_capacity(32 * calldata.len())` and then
-        // extend the `Vec` with each `Felt::to_bytes_le()`
-        // when this `Vec<u8>` makes it to `StarknetTransport`,
-        // reconstruct the `Vec<Felt>` from it
-        todo!()
+        // Derive ecdsa key from private key
+        let secret_scalar = Felt::from_bytes_be(&self.signing_key);
+        let signing_key = StarknetSigningKey::from_secret_scalar(secret_scalar);
+        let verifying_key = signing_key.verifying_key().scalar();
+        let verifying_key_bytes = verifying_key.to_bytes_be();
+
+        // Derive ed25519 key from private key
+        let user_key = SigningKey::from_bytes(&self.signing_key).verifying_key();
+        let user_key_bytes = user_key.to_bytes();
+
+        // Create Repr wrapped ContextIdentity instances
+        let signer_id = verifying_key_bytes
+            .rt::<ContextIdentity>()
+            .map_err(|e| eyre::eyre!("Failed to convert verifying key: {}", e))?;
+        let signer_id = Repr::new(signer_id);
+
+        let user_id = user_key_bytes
+            .rt::<ContextIdentity>()
+            .map_err(|e| eyre::eyre!("Failed to convert user key: {}", e))?;
+        let user_id = Repr::new(user_id);
+
+        // Create the Request structure using into() conversions
+        let request = StarknetRequest {
+            signer_id: signer_id.into(),
+            user_id: user_id.into(),
+            nonce: 0,
+            kind: self.kind.into(),
+        };
+
+        // Serialize the request
+        let mut serialized_request = vec![];
+        request.encode(&mut serialized_request)?;
+
+        // Hash the serialized request
+        let hash = poseidon_hash_many(&serialized_request);
+
+        // Sign the hash with the signing key
+        let signature = signing_key.sign(&hash)?;
+
+        let signed_request = StarknetSigned {
+            payload: serialized_request,
+            signature_r: signature.r,
+            signature_s: signature.s,
+        };
+
+        let mut signed_request_serialized = vec![];
+        signed_request.encode(&mut signed_request_serialized)?;
+
+        let bytes: Vec<u8> = signed_request_serialized
+            .iter()
+            .flat_map(|felt| felt.to_bytes_be())
+            .collect();
+
+        Ok(bytes)
     }
 
     fn decode(_response: Vec<u8>) -> eyre::Result<Self::Returns> {
-        todo!()
+        Ok(())
     }
 }
 
-impl<'a, T: Transport> ContextConfigMutateRequest<'a, T> {
+impl<'a, T: Transport + Debug> ContextConfigMutateRequest<'a, T> {
     pub async fn send(self, signing_key: [u8; 32]) -> Result<(), ClientError<T>> {
         let request = Mutate {
             signing_key,
+            nonce: 0,
             // todo! when nonces are implemented in context
             // todo! config contract, we fetch it here first
             // nonce: _,
```

Comment:
```suggestion
            // todo! when nonces are implemented in context
            // todo! config contract, we fetch it here first
            nonce: 0,
```

SN contract doesn't validate against incremental nonces?

Reply:
It does but its in draft, since we are waiting for NEAR implementation